### PR TITLE
[ts_convert] Update conversion for aten.tensor

### PR DESCRIPTION
**Pull request by angelayi**
Fixes aten::tensor issues in edgeml models
P1492137675
| suite   |   #models |   #has_ts_model |   #has_sample_inputs |   #ts_can_run |   #can_convert |   #ep_result_correct |   #can_package |   #sigmoid_can_run |   #sigmoid_result_correct |
|---------|-----------|-----------------|----------------------|---------------|----------------|----------------------|----------------|--------------------|---------------------------|
| EDGEML  |        34 |              25 |                   23 |            21 |              2 |                    2 |              2 |                  2 |                         2 |

### CODE REVIEW

**Review by jiashenC — APPROVED** [review]

**Review by SherlockNoMad — APPROVED** [review]

File: test/export/test_converter.py [hunk 1]
```diff
@@ -869,6 +869,22 @@ def forward(self, x):
         #         )[0], 1
         #     )
 
+    def test_aten_tensor_dtype_int(self):
+        class M(torch.nn.Module):
+            def forward(self, x):
+                y = torch.tensor(1, dtype=torch.int32)
```
Comment (SherlockNoMad) [inline test/export/test_converter.py:875]: Can we also test the following case

```
s = x.shape[0]
y = torch.tensor([s, s*2, 1])
```

File: torch/_export/converter.py [hunk 2]
```diff
@@ -411,16 +413,19 @@ def convert_graph_inputs(self):
...
+            if "dtype" in kwargs and kwargs["dtype"] is not None:
+                kwargs["dtype"] = _TORCH_ENUM_TO_DTYPE[kwargs["dtype"]]
+            return torch.tensor(*args, **kwargs)
 
         output_name = node.output().debugName()
-        alias_name = f"lifted_tensor_{output_name}"
-        fx_node = self.fx_graph.get_attr(alias_name)
+        fx_node = self.fx_graph.call_function(to_tensor, args, kwargs)
         self.name_to_node[output_name] = fx_node
-        self.name_to_tensor_constants[alias_name] = tensor
```
Comment (SherlockNoMad) [inline torch/_export/converter.py:423]: I wonder if we need to special handle the case where value is all constant?

e.g. 
torch.tensor([1,2,3])
vs.
s = shape of some tensor
torch.tensor([s, s*2])

File: test/export/test_converter.py (hunk 1 above)
Comment (angelayi) [reply test/export/test_converter.py:877]: Updated with special casing depending on if the input is all constant ints or not. If it is all constant, we will replace with `torch.tensor` which will eventually create a constant tensor in the exported program. Otherwise, we will replace with `torch._ref.tensor` which will decompose into aten.scalar_tensor + aten._to_copy + aten.detach.
